### PR TITLE
Enhance broker support 

### PR DESCRIPTION
**Pull request by amirreza8002**
this is the initial version, there are more things that need to be done
i'll update the docs and README after i'm sure the code works

i've tried to make valkey an optional dependency,
loosen up redis so other servers can be configured easier
and changed some names to be more abstract

waiting for some feedback 👐🏼

related issue: #189

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -47,7 +47,8 @@ croniter = ">=2.0"
 click = "^8.1"
 rq = "^1.16"
 pyyaml = { version = "^6.0", optional = true }
-valkey = "6.0.1"
+valkey = { version = "^6.0.2", optional = true}
+libvalkey = { version = "^4.0.0", optional = true}
```
Comment (cunla) [inline pyproject.toml:51]: why is this needed?

File: scheduler/connection_types.py [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
+    valkey = redis
+    valkey.Valkey = redis.Redis
 
 from scheduler.settings import Broker
 
 ConnectionErrorType = Union[redis.ConnectionError, valkey.ConnectionError]
 ConnectionType = Union[redis.Redis, valkey.Valkey]
 PipelineType = Union[redis.client.Pipeline, valkey.client.Pipeline]
 RedisSentinel = redis.sentinel.Sentinel
+ValkeySentinel = valkey.sentinel.Sentinel
```
Comment (cunla) [inline scheduler/connection_types.py:16]: Maybe have `SentinelType = Union[...]` like `ConnectionType` and `ConnectionErrorType`

File: scheduler/views.py [hunk 3]
```diff
@@ -14,12 +13,22 @@
...
+from .queues import get_all_workers, get_connection, QueueNotFoundError, ConnectionErrors
 from .queues import get_queue as get_queue_base
 from .rq_classes import JobExecution, DjangoWorker, DjangoQueue, InvalidJobOperation
 from .settings import SCHEDULER_CONFIG, logger
 
 
+try:
+    from valkey import exceptions
+except ImportError:
+    exceptions = ""
```
Comment (cunla) [inline scheduler/views.py:26]: Perhaps move `ResponseErrors` and `ConnectionErrors` to `connection_types.py`? (and maybe rename the file to `broker_typings.py`)

**Review by cunla — COMMENTED** [review]
Hi, overall looks good and a welcomed change.

I added a few comments regarding where things should be defined.
The one thing I am not very clear on is why is `libvalkey` needed?

**Comment by amirreza8002** [comment]
I'll take a look at the comments 

libvalkey is not needed per say

it's an optional C-binding that improves performance of valkey-py
thought it's good to include, but we can remove it if you want

File: scheduler/connection_types.py (hunk 2 above)
Comment (amirreza8002) [reply scheduler/connection_types.py:16]: @cunla now i'm thinking this could also work

```
broker_client = {
    "valkey": "valkey",
    "redis": "redis",
    ...
}

Sentinel = import_string(f"{broker_client[SCHEDULER_CONFIG.BROKER.value]}.sentinel.Sentinel"}
```
same can be done for errors and...

tho it might need modification if a new broker is added that has different structure

let me know what you think

File: pyproject.toml (hunk 1 above)
Comment (cunla) [reply pyproject.toml:51]: I think let's remove it, this package is not about the internals of how the connection is made and should obfuscate it. A user can obviously add it separately.

**Comment by cunla** [comment]
Hi, I made some changes - let me know what you think.

**Comment by amirreza8002** [comment]
LGTM
I'll report if i see anything new then✌️
